msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -791,6 +795,158 @@ public static ISingleAccountPublicClientApplication createSingleAccountPublicCli
+     *                   For more information on the schema of the MSAL configuration file,
+     *                   please see <a href="https://developer.android.com/guide/topics/resources/providing-resources">Android app resource overview</a>
+     *                   and <a href="https://github.com/AzureAD/microsoft-authentication-library-for-android/wiki">MSAL Github Wiki</a>
+     *                   </p>
+     * @return An instance of INativeAuthPublicClientApplication.
+     */
+    public static INativeAuthPublicClientApplication createNativeAuthPublicClientApplication(
+            @NonNull final Context context,
+            @NonNull final File configFile) throws InterruptedException, MsalException {
+        validateNonNullArgument(context, NONNULL_CONSTANTS.CONTEXT);
+        validateNonNullArgument(configFile, NONNULL_CONSTANTS.CONFIG_FILE);
+
+        try {
+            return createNativeAuthApplication(
+                    Companion.initializeNativeAuthConfiguration(context, configFile),
+                    null,
+                    null,
+                    null,
+                    null
+            );
+        } catch (BaseException e) {
+            throw new MsalClientException(
+                    UNKNOWN_ERROR,
+                    NATIVE_AUTH_APPLICATION_CREATION_UNKNOWN_ERROR_MESSAGE,
+                    e
+            );
+        }
+    }
+
+    /**
+     * {@link PublicClientApplication#createNativeAuthPublicClientApplication(Context, String, String, String, List<String>)}
+     * will read the client id and other configuration settings from the file included in your
+     * applications resources.
+     *
+     * <p><p>This function will pass back an {@link MsalClientException} object if it is unable
+     * to return {@link INativeAuthPublicClientApplication} through the listener. For example, AccountMode
+     * in configuration is not set to single. </p></p>
+     *
+     * @param context     Application's {@link Context}. The sdk requires the application context
+     *                    to be passed in {@link PublicClientApplication}. Cannot be null.
+     *                    <p>
+     *                    Note: The {@link Context} should be the application context instead of
+     *                    the running activity's context, which could potentially make the sdk hold a
+     *                    strong reference to the activity, thus preventing correct garbage
+     *                    collection and causing bugs.
+     *                    </p>
+     * @param clientId    The application client id. Cannot be null.
+     * @param authority   The default authority to be used for the authority. If this is null, the default authority will be used.
+     * @param redirectUri The redirect URI of the application.
+     * @return An instance of INativeAuthPublicClientApplication.
+     */
+    public static INativeAuthPublicClientApplication createNativeAuthPublicClientApplication(
```

Comment:
I see you are adding a lot of overloads but do you actually need them? I think following the same pattern as existing MSAL code sounds good but you should consider whether you really have a need to provide all these? Any code you write is something you will have to support for a long time so you should consider that since you are offering a new feature then can you choose to provide just one or max 2 constructors so that you only have to supported those limited number of constructors in the future.

Reply:
Is there a reason for that aim? Do you have customer demand for these overloads?